diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -294,10 +294,12 @@
     string( TOLOWER "${ONEMATH_SYCL_IMPLEMENTATION}" ONEMATH_SYCL_IMPLEMENTATION)
     if (ONEMATH_SYCL_IMPLEMENTATION STREQUAL "hipsycl")
       message(STATUS "Looking for hipSYCL")
-      find_package(hipSYCL CONFIG REQUIRED)
+      find_package(AdaptiveCpp CONFIG REQUIRED)
       set(USE_ADD_SYCL_TO_TARGET_INTEGRATION true)
       set (CMAKE_CXX_STANDARD 17)
       add_library(ONEMATH::SYCL::SYCL INTERFACE IMPORTED)
+      target_link_libraries(ONEMATH::SYCL::SYCL INTERFACE AdaptiveCpp::acpp-rt)
+      target_compile_definitions(ONEMATH::SYCL::SYCL INTERFACE __HIPSYCL__)
     elseif(ONEMATH_SYCL_IMPLEMENTATION STREQUAL "dpc++")
       message(STATUS "Looking for dpc++")
       set(USE_ADD_SYCL_TO_TARGET_INTEGRATION false)
