diff --git a/cpp/src/parquet/size_statistics.cc b/cpp/src/parquet/size_statistics.cc
index 1ce6c937a..e45eef3f0 100644
--- a/cpp/src/parquet/size_statistics.cc
+++ b/cpp/src/parquet/size_statistics.cc
@@ -18,9 +18,11 @@
 #include "parquet/size_statistics.h"

 #include <algorithm>
+#include <array>
 #include <numeric>
 #include <ostream>
 #include <string_view>
+#include <vector>

 #include "arrow/util/logging.h"
 #include "parquet/exception.h"
