--- a/win/makefile.vc
+++ b/win/makefile.vc
@@ -741,7 +741,7 @@
 	@$(CPY) "$(TKSTUBLIB)" "$(LIB_INSTALL_DIR)\"
 !if !$(STATIC_BUILD)
 	@echo creating package index
-	@type << > $(OUT_DIR)\pkgIndex.tcl
+	@type <<$(OUT_DIR)\pkgIndex.tcl
 if {![package vsatisfies [package provide Tcl] 8.7-]} return
 if {($$::tcl_platform(platform) eq "unix") && ([info exists ::env(DISPLAY)]
 	|| ([info exists ::argv] && ("-display" in $$::argv)))} {
