--- a/octomap/CMakeLists.txt
+++ b/octomap/CMakeLists.txt
@@ -30,9 +30,7 @@
 ENDIF(DEFINED ENV{OCTOMAP_OMP})
 IF(OCTOMAP_OMP)
   FIND_PACKAGE( OpenMP REQUIRED)
-  SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-  SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
+  link_libraries(OpenMP::OpenMP_CXX)
 ENDIF(OCTOMAP_OMP)
 
 # Set output directories for libraries and executables
