--- a/glomap/CMakeLists.txt
+++ b/glomap/CMakeLists.txt
@@ -97,7 +97,6 @@
 else()
     target_compile_options(glomap PRIVATE
         -Wall
-        -Werror
         -Wno-sign-compare
         -Wno-unused-variable
     )
@@ -113,8 +112,19 @@
 )
 target_link_libraries(glomap_main glomap)
 
+include(GNUInstallDirs)
 set_target_properties(glomap_main PROPERTIES OUTPUT_NAME glomap)
-install(TARGETS glomap_main DESTINATION bin)
+install(TARGETS glomap_main DESTINATION ${CMAKE_INSTALL_BINDIR})
+
+install(TARGETS glomap
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/"
+    DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/glomap"
+    FILES_MATCHING PATTERN "*.h"
+)
 
 if(TESTS_ENABLED)
     add_executable(glomap_test
