From d2b1e3610271f6349090c4609ebf2698825ab1ee Mon Sep 17 00:00:00 2001
From: Martin Valgur <martin.valgur@gmail.com>
Date: Fri, 19 Jul 2024 17:01:21 +0300
Subject: [PATCH] Fix a typo in a CMake option name

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dff971a..6a6cee0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,7 +12,7 @@ option(ENABLE_IMUVAL "Enable IMU calibration validator" ON)
 option(ENABLE_SCAN_CONTEXT "Enable ScanContext Loop Detector" OFF)
 option(ENABLE_DBOW "Enable DBoW Loop Detector" OFF)
 option(ENABLE_GNSS "Enable GNSS module (ROS2 is not supported)" ON)
-option(ENABLE_FLATEATHER "Enable flat earth assumption module" ON)
+option(ENABLE_FLATEARTHER "Enable flat earth assumption module" ON)
 
 
 find_package(glim REQUIRED)
@@ -67,7 +67,7 @@ if(ENABLE_GNSS)
   list(APPEND glim_ext_LIBRARIES gnss_global)
 endif()
 
-if(ENABLE_FLATEATHER)
+if(ENABLE_FLATEARTHER)
   add_subdirectory(modules/mapping/flat_earther)
   list(APPEND glim_ext_LIBRARIES flat_earther)
 endif()
