diff --git a/debugbreak.h b/debugbreak.h
index 5980b77..a41d42b 100644
--- a/debugbreak.h
+++ b/debugbreak.h
@@ -118,6 +118,17 @@ __inline__ static void trap_instruction(void)
 	 * The workaround is the same as ARM Thumb mode: use debugbreak-gdb.py
 	 * or manually jump over the instruction. */
 }
+#elif defined(__riscv)
+	/* RISC-V 32 or 64-bit, whether the "C" extension
+	 * for compressed, 16-bit instructions are supported or not */
+	#define DEBUG_BREAK_IMPL DEBUG_BREAK_USE_TRAP_INSTRUCTION
+__attribute__((always_inline))
+__inline__ static void trap_instruction(void)
+{
+	/* See 'riscv-tdep.c' in GDB source,
+	 * 'riscv_sw_breakpoint_from_kind' */
+	__asm__ volatile(".4byte 0x00100073");
+}
 #else
 	#define DEBUG_BREAK_IMPL DEBUG_BREAK_USE_SIGTRAP
 #endif
