--- src/dft/CMakeLists.txt
+++ src/dft/CMakeLists.txt
@@ -43,9 +43,10 @@
   set(COMMON_TARGET_DEFINITIONS ${COMMON_TARGET_DEFINITIONS} ENABLE_LONGDOUBLE)
 endif (COMPILER_SUPPORTS_LONG_DOUBLE)
 
-if(COMPILER_SUPPORTS_OPENMP)
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-endif(COMPILER_SUPPORTS_OPENMP)
+if(NOT DISABLE_OPENMP)
+  find_package(OpenMP REQUIRED)
+  link_libraries(OpenMP::OpenMP_C)
+endif()
 
 
 # Include directories
