--- src/dft/CMakeLists.txt
+++ src/dft/CMakeLists.txt
@@ -216,8 +216,10 @@
 
 # Compiler properties
 
-set(CMAKE_C_FLAGS "${ORG_CMAKE_C_FLAGS} ${DFT_C_FLAGS} ${OpenMP_C_FLAGS}")
-set(CMAKE_CXX_FLAGS "${ORG_CMAKE_C_FLAGS} ${DFT_C_FLAGS} ${OpenMP_C_FLAGS}")
+if(NOT SLEEF_DISABLE_OPENMP)
+  find_package(OpenMP REQUIRED)
+  link_libraries(OpenMP::OpenMP_C)
+endif()
 
 if(MSVC)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
