diff --git a/CMakeLists.txt b/CMakeLists.txt
index cd263ced2c..1ebe211a13 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,6 +107,26 @@ include("cmake/testing.cmake")
 include("cmake/host_compiler.cmake")
 include("cmake/configuring_primitive_list.cmake")
 
+find_package(level-zero REQUIRED)
+link_libraries(level-zero::level-zero)
+
+find_package(spdlog REQUIRED)
+link_libraries(spdlog::spdlog)
+
+if(DNNL_TARGET_ARCH STREQUAL "X64" OR DNNL_TARGET_ARCH STREQUAL "AARCH64")
+    if(DNNL_ENABLE_JIT_PROFILING OR DNNL_ENABLE_ITT_TASKS)
+        find_package(ittapi REQUIRED)
+        link_libraries(ittapi::ittnotify)
+    endif()
+endif()
+if(DNNL_TARGET_ARCH STREQUAL "X64")
+    find_package(xbyak REQUIRED)
+    link_libraries(xbyak::xbyak)
+elseif(DNNL_TARGET_ARCH STREQUAL "AARCH64")
+    find_package(xbyak_aarch64 REQUIRED)
+    link_libraries(xbyak_aarch64::xbyak_aarch64)
+endif()
+
 if(UNIX OR MINGW)
     # CMAKE_<lang>_STANDARD support, so set it to our defaults, unless
     # overridden by the user
diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index 25adc00e61..d1a86e00b2 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -20,7 +20,7 @@ file(GLOB SOURCES
     ${CMAKE_CURRENT_SOURCE_DIR}/*.c
     ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp
     )
-if(DNNL_ENABLE_JIT_PROFILING OR DNNL_ENABLE_ITT_TASKS)
+if(FALSE)
     if(DNNL_TARGET_ARCH STREQUAL "AARCH64" OR DNNL_TARGET_ARCH STREQUAL "X64")
         file(GLOB ITT_SOURCES
             ${PROJECT_SOURCE_DIR}/third_party/ittnotify/*.c
diff --git a/src/common/ittnotify.cpp b/src/common/ittnotify.cpp
index a8a409fb20..76d9c53c96 100644
--- a/src/common/ittnotify.cpp
+++ b/src/common/ittnotify.cpp
@@ -19,7 +19,7 @@
 
 #if defined(DNNL_ENABLE_ITT_TASKS)
 #include "dnnl_debug.h"
-#include "ittnotify/ittnotify.h"
+#include "ittnotify.h"
 #endif
 
 namespace dnnl {
diff --git a/src/cpu/aarch64/CMakeLists.txt b/src/cpu/aarch64/CMakeLists.txt
index 75ba8e2705..e3722e259b 100644
--- a/src/cpu/aarch64/CMakeLists.txt
+++ b/src/cpu/aarch64/CMakeLists.txt
@@ -41,5 +41,3 @@ set(OBJ_LIB ${LIB_PACKAGE_NAME}_cpu_aarch64)
 add_library(${OBJ_LIB} OBJECT ${SOURCES})
 set_property(GLOBAL APPEND PROPERTY DNNL_LIB_DEPS
     $<TARGET_OBJECTS:${OBJ_LIB}>)
-
-add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/xbyak_aarch64 xbyak_aarch64)
diff --git a/src/cpu/aarch64/cpu_isa_traits.hpp b/src/cpu/aarch64/cpu_isa_traits.hpp
--- a/src/cpu/aarch64/cpu_isa_traits.hpp
+++ b/src/cpu/aarch64/cpu_isa_traits.hpp
@@ -31,8 +31,8 @@
 #define XBYAK_USE_MMAP_ALLOCATOR
 #endif
 
-#include "xbyak_aarch64/xbyak_aarch64/xbyak_aarch64.h"
-#include "xbyak_aarch64/xbyak_aarch64/xbyak_aarch64_util.h"
+#include "xbyak_aarch64/xbyak_aarch64.h"
+#include "xbyak_aarch64/xbyak_aarch64_util.h"
 
 namespace dnnl {
 namespace impl {
diff --git a/src/cpu/jit_utils/jit_utils.cpp b/src/cpu/jit_utils/jit_utils.cpp
index d4a5119967..a2c92f253f 100644
--- a/src/cpu/jit_utils/jit_utils.cpp
+++ b/src/cpu/jit_utils/jit_utils.cpp
@@ -31,7 +31,7 @@
 #endif
 
 #if DNNL_ENABLE_JIT_PROFILING
-#include "ittnotify/jitprofiling.h"
+#include "jitprofiling.h"
 #ifdef __linux__
 #include "cpu/jit_utils/linux_perf/linux_perf.hpp"
 #endif
