--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,8 +31,8 @@
 
 find_package(Boost REQUIRED COMPONENTS graph filesystem)
 find_package(GTSAM REQUIRED)
-find_package(GTSAM_UNSTABLE REQUIRED)
 find_package(Eigen3 REQUIRED)
+find_package(nanoflann REQUIRED)
 
 if(BUILD_WITH_TBB)
   find_package(TBB REQUIRED)
@@ -202,7 +202,6 @@
 )
 target_include_directories(gtsam_points PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/nanoflann/include>
   $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
   $<INSTALL_INTERFACE:include>
 )
@@ -215,6 +214,7 @@
   gtsam_unstable
   $<TARGET_NAME_IF_EXISTS:TBB::tbb>
   $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX>
+  nanoflann::nanoflann
 )
 
 # GPU-related
@@ -262,6 +262,7 @@
     gtsam_unstable
     $<TARGET_NAME_IF_EXISTS:TBB::tbb>
     $<TARGET_NAME_IF_EXISTS:OpenMP::OpenMP_CXX>
+    nanoflann::nanoflann
   )
 
   target_link_libraries(gtsam_points
@@ -355,7 +356,7 @@
 
 include(GNUInstallDirs)
 install(
-  DIRECTORY include/ thirdparty/nanoflann/include/ ${CMAKE_CURRENT_BINARY_DIR}/include/
+  DIRECTORY include/ ${CMAKE_CURRENT_BINARY_DIR}/include/
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
 
