diff --git a/RenderStateNotation/CMakeLists.txt b/RenderStateNotation/CMakeLists.txt
--- a/RenderStateNotation/CMakeLists.txt	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/RenderStateNotation/CMakeLists.txt	(date 1745054235085)
@@ -91,7 +91,7 @@
     interface
 PRIVATE
     include
-    ../../DiligentCore/Graphics/Archiver/interface
+
     ${RSN_PARSER_HEADERS_DIR}
 )
 
diff --git a/Imgui/interface/ImGuiUtils.hpp b/Imgui/interface/ImGuiUtils.hpp
--- a/Imgui/interface/ImGuiUtils.hpp	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/Imgui/interface/ImGuiUtils.hpp	(date 1745054235090)
@@ -33,7 +33,7 @@
 #include <memory>
 #include <vector>
 
-#include "../../../DiligentCore/Platforms/Basic/interface/DebugUtilities.hpp"
+#include "Platforms/Basic/interface/DebugUtilities.hpp"
 
 namespace ImGui
 {
diff --git a/Imgui/interface/ImGuiImplDiligent.hpp b/Imgui/interface/ImGuiImplDiligent.hpp
--- a/Imgui/interface/ImGuiImplDiligent.hpp	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/Imgui/interface/ImGuiImplDiligent.hpp	(date 1745054235101)
@@ -28,7 +28,7 @@
 #pragma once
 
 #include <memory>
-#include "../../../DiligentCore/Primitives/interface/BasicTypes.h"
+#include "Primitives/interface/BasicTypes.h"
 
 namespace Diligent
 {
diff --git a/Imgui/interface/ImGuiDiligentRenderer.hpp b/Imgui/interface/ImGuiDiligentRenderer.hpp
--- a/Imgui/interface/ImGuiDiligentRenderer.hpp	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/Imgui/interface/ImGuiDiligentRenderer.hpp	(date 1745054235107)
@@ -28,10 +28,10 @@
 #pragma once
 
 #include <memory>
-#include "../../../DiligentCore/Primitives/interface/BasicTypes.h"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypes.h"
+#include "Primitives/interface/BasicTypes.h"
+#include "Common/interface/BasicMath.hpp"
+#include "Common/interface/RefCntAutoPtr.hpp"
+#include "Graphics/GraphicsEngine/interface/GraphicsTypes.h"
 #include "imgui.h"
 
 struct ImDrawData;
diff --git a/RenderStateNotation/interface/RenderStateNotationLoader.h b/RenderStateNotation/interface/RenderStateNotationLoader.h
--- a/RenderStateNotation/interface/RenderStateNotationLoader.h	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/RenderStateNotation/interface/RenderStateNotationLoader.h	(date 1745054235116)
@@ -31,12 +31,12 @@
 /// \file
 /// Defines Diligent::IRenderStateNotationLoader interface
 #include "RenderStateNotationParser.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/RenderStateCache.h"
+#include "Graphics/GraphicsTools/interface/RenderStateCache.h"
 
 
 DILIGENT_BEGIN_NAMESPACE(Diligent)
 
-#include "../../../DiligentCore/Primitives/interface/DefineRefMacro.h"
+#include "Primitives/interface/DefineRefMacro.h"
 
 /// Render state notation loader initialization info.
 struct RenderStateNotationLoaderCreateInfo
@@ -253,13 +253,13 @@
 
 // clang-format on
 
-#include "../../../DiligentCore/Primitives/interface/UndefRefMacro.h"
+#include "Primitives/interface/UndefRefMacro.h"
 
 // {FD9B12C5-3BC5-4729-A2B4-924DF374B3D3}
 static DILIGENT_CONSTEXPR INTERFACE_ID IID_RenderStateNotationLoader = {0xFD9B12C5, 0x3BC5, 0x4729, {0xA2, 0xB4, 0x92, 0x4D, 0xF3, 0x74, 0xB3, 0xD3}};
 
 #define DILIGENT_INTERFACE_NAME IRenderStateNotationLoader
-#include "../../../DiligentCore/Primitives/interface/DefineInterfaceHelperMacros.h"
+#include "Primitives/interface/DefineInterfaceHelperMacros.h"
 
 #define IRenderStateNotationLoaderInclusiveMethods \
     IObjectInclusiveMethods;                       \
@@ -326,7 +326,7 @@
 };
 DILIGENT_END_INTERFACE
 
-#include "../../../DiligentCore/Primitives/interface/UndefInterfaceHelperMacros.h"
+#include "Primitives/interface/UndefInterfaceHelperMacros.h"
 
 #if DILIGENT_C_INTERFACE
 
@@ -340,11 +340,11 @@
 
 #endif
 
-#include "../../../DiligentCore/Primitives/interface/DefineGlobalFuncHelperMacros.h"
+#include "Primitives/interface/DefineGlobalFuncHelperMacros.h"
 
 void DILIGENT_GLOBAL_FUNCTION(CreateRenderStateNotationLoader)(const RenderStateNotationLoaderCreateInfo REF CreateInfo,
                                                                IRenderStateNotationLoader**                  ppLoader);
 
-#include "../../../DiligentCore/Primitives/interface/UndefGlobalFuncHelperMacros.h"
+#include "Primitives/interface/UndefGlobalFuncHelperMacros.h"
 
 DILIGENT_END_NAMESPACE // namespace Diligent
diff --git a/BuildTools/FormatValidation/validate_format_win.bat b/BuildTools/FormatValidation/validate_format_win.bat
--- a/BuildTools/FormatValidation/validate_format_win.bat	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/BuildTools/FormatValidation/validate_format_win.bat	(date 1745054235121)
@@ -1,5 +1,5 @@
-python ../../../DiligentCore/BuildTools/FormatValidation/clang-format-validate.py --color never ^
---clang-format-executable ../../../DiligentCore/BuildTools/FormatValidation/clang-format_10.0.0.exe ^
+python BuildTools/FormatValidation/clang-format-validate.py --color never ^
+--clang-format-executable BuildTools/FormatValidation/clang-format_10.0.0.exe ^
 -r ../../AssetLoader ../../Imgui ../../NativeApp/include  ../../NativeApp/src ../../TextureLoader ../../Tests ^
 ../../RenderStateNotation ../../RenderStatePackager ^
 --exclude ../../Imgui/interface/ImGuiImplMacOS.h ^
diff --git a/RenderStateNotation/interface/RenderStateNotationParser.h b/RenderStateNotation/interface/RenderStateNotationParser.h
--- a/RenderStateNotation/interface/RenderStateNotationParser.h	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/RenderStateNotation/interface/RenderStateNotationParser.h	(date 1745054235130)
@@ -30,7 +30,7 @@
 
 /// \file
 /// Defines Diligent::IRenderStateNotationParser interface
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "Graphics/GraphicsEngine/interface/RenderDevice.h"
 
 DILIGENT_BEGIN_NAMESPACE(Diligent)
 
@@ -392,7 +392,7 @@
 static DILIGENT_CONSTEXPR INTERFACE_ID IID_RenderStateNotationParser = {0x355AC9F7, 0x5D9D, 0x423D, {0xAE, 0x35, 0x80, 0xE0, 0x02, 0x8D, 0xE1, 0x7E}};
 
 #define DILIGENT_INTERFACE_NAME IRenderStateNotationParser
-#include "../../../DiligentCore/Primitives/interface/DefineInterfaceHelperMacros.h"
+#include "Primitives/interface/DefineInterfaceHelperMacros.h"
 
 #define IRenderStateNotationParserInclusiveMethods \
     IObjectInclusiveMethods;                       \
@@ -555,7 +555,7 @@
 };
 DILIGENT_END_INTERFACE
 
-#include "../../../DiligentCore/Primitives/interface/UndefInterfaceHelperMacros.h"
+#include "Primitives/interface/UndefInterfaceHelperMacros.h"
 
 #if DILIGENT_C_INTERFACE
 
@@ -577,12 +577,12 @@
 
 #endif
 
-#include "../../../DiligentCore/Primitives/interface/DefineGlobalFuncHelperMacros.h"
+#include "Primitives/interface/DefineGlobalFuncHelperMacros.h"
 
 void DILIGENT_GLOBAL_FUNCTION(CreateRenderStateNotationParser)(const RenderStateNotationParserCreateInfo REF CreateInfo,
                                                                IRenderStateNotationParser**                  pParser);
 
 
-#include "../../../DiligentCore/Primitives/interface/UndefGlobalFuncHelperMacros.h"
+#include "Primitives/interface/UndefGlobalFuncHelperMacros.h"
 
 DILIGENT_END_NAMESPACE // namespace Diligent
diff --git a/BuildTools/FormatValidation/validate_format_linux.sh b/BuildTools/FormatValidation/validate_format_linux.sh
--- a/BuildTools/FormatValidation/validate_format_linux.sh	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/BuildTools/FormatValidation/validate_format_linux.sh	(date 1745054235135)
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-source ../../../DiligentCore/BuildTools/FormatValidation/validate_format_linux_implementation.sh
+source BuildTools/FormatValidation/validate_format_linux_implementation.sh
 
 validate_format ../../AssetLoader ../../Imgui ../../NativeApp/include  ../../NativeApp/src ../../TextureLoader ../../Tests \
 ../../RenderStateNotation ../../RenderStatePackager \
diff --git a/Tests/DiligentToolsGPUTest/CMakeLists.txt b/Tests/DiligentToolsGPUTest/CMakeLists.txt
--- a/Tests/DiligentToolsGPUTest/CMakeLists.txt	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/Tests/DiligentToolsGPUTest/CMakeLists.txt	(date 1745054235138)
@@ -4,7 +4,7 @@
 
 file(GLOB_RECURSE INCLUDE include/*.*)
 file(GLOB_RECURSE SOURCE src/*.*)
-set(MAIN_CPP ../../../DiligentCore/Tests/DiligentCoreAPITest/src/main.cpp)
+set(MAIN_CPP Tests/DiligentCoreAPITest/src/main.cpp)
 file(GLOB_RECURSE SHADERS assets/Shaders/*.*)
 file(GLOB_RECURSE RENDER_STATES assets/RenderStates/*.*)
 
diff --git a/AssetLoader/interface/GLTFResourceManager.hpp b/AssetLoader/interface/GLTFResourceManager.hpp
--- a/AssetLoader/interface/GLTFResourceManager.hpp	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/AssetLoader/interface/GLTFResourceManager.hpp	(date 1745054235144)
@@ -32,13 +32,13 @@
 #include <unordered_map>
 #include <atomic>
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/ObjectBase.hpp"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/BufferSuballocator.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/DynamicTextureAtlas.h"
-#include "../../../DiligentCore/Graphics/GraphicsTools/interface/VertexPoolX.hpp"
+#include "Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "Common/interface/RefCntAutoPtr.hpp"
+#include "Common/interface/ObjectBase.hpp"
+#include "Graphics/GraphicsTools/interface/BufferSuballocator.h"
+#include "Graphics/GraphicsTools/interface/DynamicTextureAtlas.h"
+#include "Graphics/GraphicsTools/interface/VertexPoolX.hpp"
 
 namespace Diligent
 {
diff --git a/AssetLoader/interface/DXSDKMeshLoader.hpp b/AssetLoader/interface/DXSDKMeshLoader.hpp
--- a/AssetLoader/interface/DXSDKMeshLoader.hpp	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/AssetLoader/interface/DXSDKMeshLoader.hpp	(date 1745054235148)
@@ -29,14 +29,14 @@
 
 #include <vector>
 
-#include "../../../DiligentCore/Primitives/interface/BasicTypes.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/Buffer.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/Texture.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/TextureView.h"
-#include "../../../DiligentCore/Common/interface/BasicMath.hpp"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#include "Primitives/interface/BasicTypes.h"
+#include "Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "Graphics/GraphicsEngine/interface/Buffer.h"
+#include "Graphics/GraphicsEngine/interface/Texture.h"
+#include "Graphics/GraphicsEngine/interface/TextureView.h"
+#include "Common/interface/BasicMath.hpp"
+#include "Common/interface/RefCntAutoPtr.hpp"
 
 namespace Diligent
 {
diff --git a/AssetLoader/interface/GLTFLoader.hpp b/AssetLoader/interface/GLTFLoader.hpp
--- a/AssetLoader/interface/GLTFLoader.hpp	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/AssetLoader/interface/GLTFLoader.hpp	(date 1745054235155)
@@ -38,13 +38,13 @@
 #include <limits>
 #include <algorithm>
 
-#include "../../../DiligentCore/Platforms/interface/PlatformMisc.hpp"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/DeviceContext.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
-#include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#include "../../../DiligentCore/Common/interface/AdvancedMath.hpp"
-#include "../../../DiligentCore/Common/interface/STDAllocator.hpp"
+#include "Platforms/interface/PlatformMisc.hpp"
+#include "Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "Graphics/GraphicsEngine/interface/DeviceContext.h"
+#include "Graphics/GraphicsEngine/interface/GraphicsTypesX.hpp"
+#include "Common/interface/RefCntAutoPtr.hpp"
+#include "Common/interface/AdvancedMath.hpp"
+#include "Common/interface/STDAllocator.hpp"
 #include "GLTFResourceManager.hpp"
 
 namespace tinygltf
diff --git a/Tests/DiligentToolsTest/CMakeLists.txt b/Tests/DiligentToolsTest/CMakeLists.txt
--- a/Tests/DiligentToolsTest/CMakeLists.txt	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/Tests/DiligentToolsTest/CMakeLists.txt	(date 1745054235161)
@@ -4,7 +4,7 @@
 
 file(GLOB_RECURSE INCLUDE include/*.*)
 file(GLOB_RECURSE SOURCE src/*.*)
-set(MAIN_CPP ../../../DiligentCore/Tests/DiligentCoreTest/src/main.cpp)
+set(MAIN_CPP Tests/DiligentCoreTest/src/main.cpp)
 file(GLOB_RECURSE ASSETS assets/*.*)
 
 if (NOT ARCHIVER_SUPPORTED)
diff --git a/TextureLoader/interface/TextureUtilities.h b/TextureLoader/interface/TextureUtilities.h
--- a/TextureLoader/interface/TextureUtilities.h	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/TextureLoader/interface/TextureUtilities.h	(date 1745054235166)
@@ -30,15 +30,15 @@
 /// \file
 /// Defines texture utilities
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypes.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/Texture.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "Graphics/GraphicsEngine/interface/GraphicsTypes.h"
+#include "Graphics/GraphicsEngine/interface/Texture.h"
+#include "Graphics/GraphicsEngine/interface/RenderDevice.h"
 #include "TextureLoader.h"
 
 DILIGENT_BEGIN_NAMESPACE(Diligent)
 
 
-#include "../../../DiligentCore/Primitives/interface/DefineGlobalFuncHelperMacros.h"
+#include "Primitives/interface/DefineGlobalFuncHelperMacros.h"
 
 /// Parameters of the CopyPixels function.
 struct CopyPixelsAttribs
@@ -166,6 +166,6 @@
                                                      IRenderDevice*            pDevice,
                                                      ITexture**                ppTexture);
 
-#include "../../../DiligentCore/Primitives/interface/UndefGlobalFuncHelperMacros.h"
+#include "Primitives/interface/UndefGlobalFuncHelperMacros.h"
 
 DILIGENT_END_NAMESPACE // namespace Diligent
diff --git a/TextureLoader/interface/Image.h b/TextureLoader/interface/Image.h
--- a/TextureLoader/interface/Image.h	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/TextureLoader/interface/Image.h	(date 1745054235172)
@@ -27,15 +27,15 @@
 
 #pragma once
 
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/GraphicsTypes.h"
-#include "../../../DiligentCore/Primitives/interface/FileStream.h"
-#include "../../../DiligentCore/Primitives/interface/DataBlob.h"
+#include "Graphics/GraphicsEngine/interface/GraphicsTypes.h"
+#include "Primitives/interface/FileStream.h"
+#include "Primitives/interface/DataBlob.h"
 
 #if DILIGENT_CPP_INTERFACE
 #    include <vector>
 
-#    include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
-#    include "../../../DiligentCore/Common/interface/ObjectBase.hpp"
+#    include "Common/interface/RefCntAutoPtr.hpp"
+#    include "Common/interface/ObjectBase.hpp"
 #endif
 
 DILIGENT_BEGIN_NAMESPACE(Diligent)
diff --git a/TextureLoader/interface/TextureLoader.h b/TextureLoader/interface/TextureLoader.h
--- a/TextureLoader/interface/TextureLoader.h	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/TextureLoader/interface/TextureLoader.h	(date 1745054235179)
@@ -27,13 +27,13 @@
 
 #pragma once
 
-#include "../../../DiligentCore/Primitives/interface/FileStream.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/RenderDevice.h"
-#include "../../../DiligentCore/Graphics/GraphicsEngine/interface/Texture.h"
+#include "Primitives/interface/FileStream.h"
+#include "Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "Graphics/GraphicsEngine/interface/Texture.h"
 #include "Image.h"
 
 #if DILIGENT_CPP_INTERFACE
-#    include "../../../DiligentCore/Common/interface/RefCntAutoPtr.hpp"
+#    include "Common/interface/RefCntAutoPtr.hpp"
 #endif
 
 DILIGENT_BEGIN_NAMESPACE(Diligent)
@@ -182,7 +182,7 @@
     {0xe04fe6d5, 0x8665, 0x4183, {0xa8, 0x72, 0x85, 0x2e, 0xf, 0x7c, 0xe2, 0x42}};
 
 #define DILIGENT_INTERFACE_NAME ITextureLoader
-#include "../../../DiligentCore/Primitives/interface/DefineInterfaceHelperMacros.h"
+#include "Primitives/interface/DefineInterfaceHelperMacros.h"
 
 #define ITextureLoaderInclusiveMethods \
     IObjectInclusiveMethods;           \
@@ -212,7 +212,7 @@
 DILIGENT_END_INTERFACE
 // clang-format on
 
-#include "../../../DiligentCore/Primitives/interface/UndefInterfaceHelperMacros.h"
+#include "Primitives/interface/UndefInterfaceHelperMacros.h"
 
 #if DILIGENT_C_INTERFACE
 
@@ -225,7 +225,7 @@
 
 #endif
 
-#include "../../../DiligentCore/Primitives/interface/DefineGlobalFuncHelperMacros.h"
+#include "Primitives/interface/DefineGlobalFuncHelperMacros.h"
 
 /// Creates a texture loader from image.
 
@@ -320,6 +320,6 @@
                                                 const TextureDesc REF Desc,
                                                 const TextureData REF TexData);
 
-#include "../../../DiligentCore/Primitives/interface/UndefGlobalFuncHelperMacros.h"
+#include "Primitives/interface/UndefGlobalFuncHelperMacros.h"
 
 DILIGENT_END_NAMESPACE // namespace Diligent
diff --git a/BuildTools/FormatValidation/validate_format_mac.sh b/BuildTools/FormatValidation/validate_format_mac.sh
--- a/BuildTools/FormatValidation/validate_format_mac.sh	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/BuildTools/FormatValidation/validate_format_mac.sh	(date 1745054235184)
@@ -1,6 +1,6 @@
 #!/bin/bash
-python3 ../../../DiligentCore/BuildTools/FormatValidation/clang-format-validate.py \
---clang-format-executable ../../../DiligentCore/BuildTools/FormatValidation/clang-format_mac_10.0.0 \
+python3 BuildTools/FormatValidation/clang-format-validate.py \
+--clang-format-executable BuildTools/FormatValidation/clang-format_mac_10.0.0 \
 -r ../../AssetLoader ../../Imgui ../../NativeApp/include  ../../NativeApp/src ../../TextureLoader ../../Tests \
 ../../RenderStateNotation ../../RenderStatePackager \
 --exclude ../../Imgui/interface/ImGuiImplMacOS.h \
diff --git a/TextureLoader/interface/BCTools.h b/TextureLoader/interface/BCTools.h
--- a/TextureLoader/interface/BCTools.h	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/TextureLoader/interface/BCTools.h	(date 1745054235189)
@@ -26,7 +26,7 @@
 
 #pragma once
 
-#include "../../../DiligentCore/Primitives/interface/BasicTypes.h"
+#include "Primitives/interface/BasicTypes.h"
 
 DILIGENT_BEGIN_NAMESPACE(Diligent)
 
diff --git a/Tests/IncludeTest/BaseStructAlignmentTest.cpp b/Tests/IncludeTest/BaseStructAlignmentTest.cpp
--- a/Tests/IncludeTest/BaseStructAlignmentTest.cpp	(revision 66b9a6ed76bc4577e43feb58a08a1c17689b6ce2)
+++ b/Tests/IncludeTest/BaseStructAlignmentTest.cpp	(date 1745054235194)
@@ -24,7 +24,7 @@
  *  of the possibility of such damages.
  */
 
-#include "../../../DiligentCore/Primitives/interface/CheckBaseStructAlignment.hpp"
+#include "Primitives/interface/CheckBaseStructAlignment.hpp"
 #include "RenderStateNotation/interface/RenderStateNotationParser.h"
 
 namespace Diligent
