--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -124,16 +124,8 @@
 endif()
 
 # Check if compiler supports OpenMP
-find_package(OpenMP)
-if (OpenMP_FOUND)
-  message(WARNING "OpenMP found! OpenMP_C_INCLUDE_DIRS = ${OpenMP_C_INCLUDE_DIRS}")
-  include_directories(${OpenMP_C_INCLUDE_DIRS})
-  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
-  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
-  set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
-else()
-  message(WARNING "OpenMP is not supported by the compiler")
-endif()
+find_package(OpenMP REQUIRED)
+link_libraries(OpenMP::OpenMP_CXX)
 
 # Define file lists
 get_filelist("get_fbgemm_generic_srcs(with_base=True)" FBGEMM_GENERIC_SRCS)
