diff --git a/cmake/kokkos_enable_devices.cmake b/cmake/kokkos_enable_devices.cmake
index c7d189285..cdbc5abb7 100644
--- a/cmake/kokkos_enable_devices.cmake
+++ b/cmake/kokkos_enable_devices.cmake
@@ -74,17 +74,8 @@
 
 kokkos_device_option(OPENMPTARGET OFF DEVICE "Whether to build the OpenMP target backend")
 if(KOKKOS_ENABLE_OPENMPTARGET)
-  set(ClangOpenMPFlag -fopenmp=libomp)
-
-  compiler_specific_flags(
-    Clang
-    ${ClangOpenMPFlag}
-    -Wno-openmp-mapping
-    -Wno-unknown-cuda-version
-    -Wno-pass-failed
-    DEFAULT
-    -fopenmp
-  )
+  find_package(OpenMP REQUIRED)
+  link_libraries(OpenMP::OpenMP_CXX)
   compiler_specific_defs(Clang KOKKOS_WORKAROUND_OPENMPTARGET_CLANG)
 endif()
 
diff --git a/cmake/kokkos_tpls.cmake b/cmake/kokkos_tpls.cmake
index 6ef3b79bd..06c48025e 100644
--- a/cmake/kokkos_tpls.cmake
+++ b/cmake/kokkos_tpls.cmake
@@ -106,7 +106,7 @@
   find_package(OpenMP 3.0 REQUIRED COMPONENTS CXX)
   kokkos_export_cmake_tpl(OpenMP REQUIRED COMPONENTS CXX)
   if(Kokkos_ENABLE_HIP AND KOKKOS_COMPILE_LANGUAGE STREQUAL HIP)
-    global_append(KOKKOS_AMDGPU_OPTIONS ${OpenMP_CXX_FLAGS})
+    link_libraries(OpenMP::OpenMP_CXX)
   endif()
   if(Kokkos_ENABLE_CUDA AND KOKKOS_COMPILE_LANGUAGE STREQUAL CUDA)
     global_append(KOKKOS_CUDA_OPTIONS -Xcompiler ${OpenMP_CXX_FLAGS})
