diff --git a/libtoolize.in b/libtoolize.in
index 0c40fed..e82e467 100644
--- a/libtoolize.in
+++ b/libtoolize.in
@@ -1903,11 +1903,12 @@ func_require_seen_libtool ()
   pkgmacro_files="@pkgmacro_files@"
 
   # Locations for important files:
-  prefix="@prefix@"
-  datadir="@datadir@"
-  pkgauxdir="@pkgauxdir@"
-  pkgltdldir="@pkgdatadir@"
-  aclocaldir="@aclocaldir@"
+  scriptdir=$(CDPATH= cd -- "$(dirname -- "$0")" && pwd)
+  prefix="$scriptdir/.."
+  datadir="$scriptdir/../share"
+  pkgauxdir="$scriptdir/../share/libtool/build-aux"
+  pkgltdldir="$scriptdir/../share/libtool"
+  aclocaldir="$scriptdir/../share/aclocal"
 
   # Allow the user to override the master libtoolize repository:
   if test -n "$_lt_pkgdatadir"; then
