--- a/libtoolize.in
+++ b/libtoolize.in
@@ -1934,11 +1934,11 @@
   pkgmacro_files="@pkgmacro_files@"
 
   # Locations for important files:
-  prefix="@prefix@"
-  datadir="@datadir@"
-  pkgauxdir="@pkgauxdir@"
-  pkgltdldir="@pkgdatadir@"
-  aclocaldir="@aclocaldir@"
+  prefix="$(CDPATH= cd -- "$(dirname -- "$0")/.." && pwd)"
+  datadir="$prefix/share"
+  pkgauxdir="$prefix/share/libtool/build-aux"
+  pkgltdldir="$prefix/share/libtool"
+  aclocaldir="$prefix/share/aclocal"
 
   # Allow the user to override the master libtoolize repository:
   if test -n "$_lt_pkgdatadir"; then
