--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,15 +18,13 @@
     set(DILIGENT_INSTALL_FX OFF)
 endif()
 
-target_link_libraries(DiligentFX 
-PRIVATE
-    Diligent-BuildSettings
+find_package(diligent-core REQUIRED CONFIG)
+find_package(diligent-tools REQUIRED CONFIG)
+
+target_link_libraries(DiligentFX
 PUBLIC
-    Diligent-GraphicsEngine
-    Diligent-GraphicsTools
-    Diligent-AssetLoader
-    Diligent-TextureLoader
-    Diligent-Imgui
+    diligent-core::diligent-core
+    diligent-tools::diligent-tools
 )
 
 if (TARGET Diligent-HLSL2GLSLConverterLib)
