--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -474,14 +474,7 @@
 ENDFOREACH(dir_types includedirs libdirs datadir docdir)
 # -----------------------------------------------------------------------------
 # OpenMP_CXX_FOUND, OpenMP_CXX_FLAGS
-IF( ${CMAKE_SYSTEM_NAME} STREQUAL "Darwin" )
-   SET(OpenMP_CXX_FOUND FALSE)
-   MESSAGE(STATUS
-      "Skipping OpenMP on Darwin (Mac) systems. 2DO: fix this"
-   )
-ELSE( ${CMAKE_SYSTEM_NAME} STREQUAL "Darwin" )
-   FIND_PACKAGE(OpenMP)
-ENDIF( ${CMAKE_SYSTEM_NAME} STREQUAL "Darwin" )
+FIND_PACKAGE(OpenMP REQUIRED)
 # -----------------------------------------------------------------------------
 # boost_prefix
 # Only need components for object libraries, not include libraries.
