--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -161,14 +161,8 @@
 # openmp
 set(USE_OPEN_MP TRUE CACHE BOOL "Set to FALSE to not use OpenMP")
 if (USE_OPEN_MP)
-	find_package(OpenMP)
-	if (OPENMP_FOUND)
-		target_compile_options(${LIB_NAME} PRIVATE -fopenmp ${OpenMP_CXX_FLAGS})
-		target_compile_definitions(${LIB_NAME} PRIVATE HAVE_OPENMP)
-		if (CMAKE_COMPILER_IS_GNUCC)
-			target_link_libraries(${LIB_NAME} PUBLIC gomp)
-		endif()
-	endif()
+	find_package(OpenMP REQUIRED)
+	target_link_libraries(${LIB_NAME} PUBLIC OpenMP::OpenMP_CXX)
 endif ()
 
 
